pkg/resourceinterpreter/configurableinterpreter/configurable.go

```diff
@@ -0,0 +1,143 @@
+package configurableinterpreter
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/klog/v2"
+
+	configv1alpha1 "github.com/karmada-io/karmada/pkg/apis/config/v1alpha1"
+	workv1alpha2 "github.com/karmada-io/karmada/pkg/apis/work/v1alpha2"
+	"github.com/karmada-io/karmada/pkg/resourceinterpreter/configurableinterpreter/configmanager"
+	"github.com/karmada-io/karmada/pkg/util/fedinformer/genericmanager"
+)
+
+// ConfigurableInterpreter interpret custom resource with resource interpreter customizations configuration
+type ConfigurableInterpreter struct {
+	// configManager caches all re configurations.
+	configManager configmanager.ConfigManager
+}
+
+// NewConfigurableInterpreter return a new ConfigurableInterpreter.
+func NewConfigurableInterpreter(informer genericmanager.SingleClusterInformerManager) (*ConfigurableInterpreter, error) {
+	return &ConfigurableInterpreter{
+		configManager: configmanager.NewInterpreterConfigManager(informer),
+	}, nil
+}
+
+// HookEnabled tells if any hook exist for specific resource gvk and operation type.
+func (c *ConfigurableInterpreter) HookEnabled(kind schema.GroupVersionKind, operationType configv1alpha1.InterpreterOperation) bool {
+	if !c.configManager.HasSynced() {
+		klog.Errorf("not yet ready to handle request")
+		return false
+	}
+	_, exist := c.configManager.LuaScriptAccessors()[kind]
+	return exist
```

Comment:
Here also need to check the `operationType`, If no script is configured for the `operationType`, the system will execute the subsequent resource interpreter.